### PR TITLE
Component test readiness polling

### PR DESCRIPTION
**Pull request by LukeLoze1**
## Motivation and Context
The current 'make test' takes a long time because:
   1. There's a couple of lengthy sleeps to make sure that the various services are up and healthy
   2. The census-pubsub image is built from scratch each time

## What has changed
   The setup_pubsub.sh now polls all 3 services to check if they're healthy and doesn't continue until they are.
   In docker-compose.yml the census-rm-pubsub app is just copied rather than a new image built 

## How to test?
Run 'make test' all should pass
## Links
https://trello.com/c/hnDKKRdj/484-tests-for-running-census-rm-pubsub-13

### CODE REVIEW

File: test/component/setup_pubsub.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
```
Comment (jcox-dev) [inline test/component/setup_pubsub.sh]:
```suggestion
#!/bin/bash
```

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Try this to fix in travis.

File: test/component/setup_pubsub.sh (hunk 1 above)
Comment (LukeLoze1) [reply test/component/setup_pubsub.sh]: done

File: test/component/setup_pubsub.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+echo 'Running setup_pubsub.sh'
+
+wait_for_curl_success() {
+    local healthcheck_url=${1}
+    local http_verb=$2
```
Comment (jcox-dev) [inline test/component/setup_pubsub.sh]: Should this also be in curly brackets?

```suggestion
    local http_verb=${2}
```

File: test/component/setup_pubsub.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+wait_for_curl_success "http://guest:guest@localhost:46672/api/aliveness-test/%2F" "GET" "rabbit_mq"
+
+echo "Waiting for [pubsub] to be ready"
+while true; do
+    response=$(docker inspect pubsub -f "{{ .State.Health.Status }}")
+    if [[ "$response" == "healthy" ]]; then
+    	echo "[pubsub] is ready"
+    	break
+	fi
```
Comment (jcox-dev) [inline test/component/setup_pubsub.sh]:
```suggestion
    fi
```

File: test/component/setup_pubsub.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        if [[ response -eq 200 ]] || [[ response -eq 409 ]]; then
+          break
+        fi
+
+        echo "[$service_name] not ready ([$response] was the response from curl)"
+        sleep 1s
+    done
+
+	echo "[$service_name] is ready"
```
Comment (jcox-dev) [inline test/component/setup_pubsub.sh]:
```suggestion
    echo "[$service_name] is ready"
```

File: readiness.patch [hunk 5]
```diff
@@ -0,0 +1,109 @@
+diff --git a/Makefile b/Makefile
```
Comment (jcox-dev) [inline readiness.patch]: File added by mistake.

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Mix of tabs and spaces in the setup script and patch file added mistakenly to branch.

File: test/component/setup_pubsub.sh (hunk 2 above)
Comment (LukeLoze1) [reply test/component/setup_pubsub.sh]: yes, though it seemed to work without it, which is interesting, for a given value of interesting

**Comment by LukeLoze1** [comment]
[tests for running census-rm-pubsub (13)](https://trello.com/c/hnDKKRdj/484-tests-for-running-census-rm-pubsub-13)

**Review by jcox-dev — APPROVED** [review]

**Review by AdamHawtin — APPROVED** [review]

**Review by hbrace — APPROVED** [review]

**Pull request merged**